### PR TITLE
S3 archive sample and clarification

### PR DESCRIPTION
**Pull request by PingDavidR**
- Closes BRASS-1217
- Demonstration-only example of S3 archive of PD backup file

### CODE REVIEW

File: 30-helm/s3-sidecar/pd-archive-backup-to-s3.yaml [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        #!/bin/sh
+
+        # Read admin password from script argument
+        if [ "$#" -ne 1 ]; then
+          echo "Expected 1 parameter: admin user password."
+          exit 1
+        fi
+        _adminPassword="${1}"
+
+        # TODO implement based on the environment: pull down backup files (from S3 or other storage) to SERVER_RESTORE_DIR
```
Comment (henryrecker-pingidentity) [inline 30-helm/s3-sidecar/pd-archive-backup-to-s3.yaml:52]: If we're going to show how the archive step works we probably should include the logic to pull from s3 here

Comment (PingDavidR) [reply 30-helm/s3-sidecar/pd-archive-backup-to-s3.yaml]: added in latest commit

File: 30-helm/s3-sidecar/pd-archive-backup-to-s3.yaml [hunk 2]
```diff
@@ -12,6 +12,20 @@
...
+        set -e
+
+        # This script assumes it is only being run on a single pod of the topology
+        # Demonstration only - no testing or error checking is performed
+
+        # Run the fetch from archive command
+        # Assumes credentials are set as environment variables or that the Pod 
+        # has an associated IAM role that allows it to write to the S3 bucket
+        # userRoot is hardcoded as it is the only backend used in this example
+        aws s3 sync s3://<your-s3-bucket-name>/<path>/userRoot/ /opt/pingdirectory-backup/userRoot
```
Comment (henryrecker-pingidentity) [inline 30-helm/s3-sidecar/pd-archive-backup-to-s3.yaml:28]: Should we pull the backups to a different folder? Is there a chance that we could have conflicting files?

Comment (PingDavidR) [reply 30-helm/s3-sidecar/pd-archive-backup-to-s3.yaml]: done - using /tmp/restore as in the restore.sh script

**Pull request merged**